src/main/webapp/scripts/perspective.js

```diff
@@ -1,21 +1,30 @@
+/*
+ *  Makes a new comment object form the text box
+ *  If it is valid, sends a Request object to Perspective API and updates DOM
+ *  If not valid, updates DOM with an error message
+ */
+
 function getToxicity() {
-  API_KEY = '';
-  const commentText = document.getElementById("comment-text").value;
-  const commentObject = {
-    'comment': {'text': commentText},
-    'requestedAttributes': {'TOXICITY': {}},
-    'languages': ['en']
-  };
-  const json = JSON.stringify(commentObject);
+  const comment = new Comment(document.getElementById("comment-text").value);
+  if(!comment.isValid) {
+    updateDom(comment.getError(), 'toxicity-container');
```

Comment:
It's kind of a shame that we need to call getError twice whenever a Comment is invalid (once in the constructor and once here). Also, every client of Comment is responsible for checking that it is valid before using it. Any ideas about how to avoid these issues?

Reply:
avoided by having the constructor throw errors instead of having a function that returns strings